Q:

Problem TurtleBot Odometry and Gyro Calibration

Hi,
i have a little problem when I launch calibration for my turtlebot :
" roslaunch turtlebot_calibration calibrate.launch "
I have this error on my screen :
  File "/home/sr1/turtlebot/src/turtlebot_apps/turtlebot_calibration/src/turtlebot_calibration/calibrate.py", line 250, in <module>
    main()
  File "/home/sr1/turtlebot/src/turtlebot_apps/turtlebot_calibration/src/turtlebot_calibration/calibrate.py", line 228, in main
    robot = CalibrateRobot()
  File "/home/sr1/turtlebot/src/turtlebot_apps/turtlebot_calibration/src/turtlebot_calibration/calibrate.py", line 69, in __init__
    self.has_gyro = rospy.get_param("turtlebot_node/has_gyro")
  File "/opt/ros/hydro/lib/python2.7/dist-packages/rospy/client.py", line 452, in get_param
    return _param_server[param_name] #MasterProxy does all the magic for us
  File "/opt/ros/hydro/lib/python2.7/dist-packages/rospy/msproxy.py", line 117, in __getitem__
    raise KeyError(key)
KeyError: 'turtlebot_node/has_gyro'
================================================================================REQUIRED process [turtlebot_calibration-15] has died!
process has died [pid 6846, exit code 1, cmd /home/sr1/turtlebot/src/turtlebot_apps/turtlebot_calibration/src/turtlebot_calibration/calibrate.py imu:=mobile_base/sensors/imu_data_raw cmd_vel:=mobile_base/commands/velocity scan_angle:=scan_angle odom:=odom __name:=turtlebot_calibration __log:=/home/sr1/.ros/log/b326776c-f689-11e3-bbee-6c881460500c/turtlebot_calibration-15.log].
log file: /home/sr1/.ros/log/b326776c-f689-11e3-bbee-6c881460500c/turtlebot_calibration-15*.log
Initiating shutdown!
================================================================================
[turtlebot_calibration-15] killing on exit
[scan_to_angle-14] killing on exit
[depthimage_to_laserscan-13] killing on exit
 [camera/disparity_registered_hw-12] killing on exit
[camera/points_xyzrgb_hw_registered-10] killing on exit
[camera/depth_registered_rectify_depth-9] killing on exit
 [camera/depth_metric_rect-6] killing on exit
[camera/rectify_ir-4] killing on exit
[camera/disparity_depth-11] killing on exit
[camera/rectify_color-3] killing on exit
[camera/depth_rectify_depth-5] killing on exit
[camera/driver-2] killing on exit
 [camera/camera_nodelet_manager-1] killing on exit
Traceback (most recent call last):
  File "/usr/lib/python2.7/site.py", line 562, in <module>
[camera/depth_points-8] killing on exit
[camera/depth_metric-7] killing on exit
    main()
  File "/usr/lib/python2.7/site.py", line 544, in main
    known_paths = addusersitepackages(known_paths)
  File "/usr/lib/python2.7/site.py", line 271, in addusersitepackages
    user_site = getusersitepackages()
  File "/usr/lib/python2.7/site.py", line 246, in getusersitepackages
    user_base = getuserbase() # this will also set USER_BASE
  File "/usr/lib/python2.7/site.py", line 235, in getuserbase
    from sysconfig import get_config_var
  File "/usr/lib/python2.7/sysconfig.py", line 124, in <module>
    _PROJECT_BASE = os.path.dirname(_safe_realpath(sys.executable))
  File "/usr/lib/python2.7/sysconfig.py", line 119, in _safe_realpath
    return realpath(path)
  File "/usr/lib/python2.7/posixpath.py", line 367, in realpath
    resolved = _resolve_link(component)
  File "/usr/lib/python2.7/posixpath.py", line 392, in _resolve_link
    dir = dirname(path)
  File "/usr/lib/python2.7/posixpath.py", line 120, in dirname
    i = p.rfind('/') + 1
KeyboardInterrupt
[camera/driver-2] escalating to SIGTERM
[camera/rectify_ir-4] escalating to SIGTERM
[camera/rectify_color-3] escalating to SIGTERM
[camera/camera_nodelet_manager-1] escalating to SIGTERM
shutting down processing monitor...
... shutting down processing monitor complete
done

Thanks for your help!

Originally posted by guigui on ROS Answers with karma: 33 on 2014-06-17
Post score: 1

Original comments
Comment by kangnamtofu on 2017-03-06:
I am having the same problem. Can someone shed a light for a noob?
Comment by TravelAmerica on 2017-03-08:
I encounter exactly same problem.
KeyError: 'turtlebot_node/has_gyro'
Any thought?
Comment by vik748 on 2017-05-25:
Same here...

A:

I manually set has_gyro = True (or False) but either stucks here.
[ INFO] [1489098043.383494668]: Initializing nodelet with 4 worker threads.
[ INFO] [1489098048.022024151]: Device "2bc5/0401@1/3" found.
Warning: USB events thread - failed to set priority. This might cause loss of data...
/opt/ros/kinetic/lib/turtlebot_calibration/scan_to_angle.py:52: SyntaxWarning: The publisher should be created with an explicit keyword argument 'queue_size'. Please see http://wiki.ros.org/rospy/Overview/Publishers%20and%20Subscribers for more information.
  self.pub = rospy.Publisher('scan_angle', ScanAngle)
[INFO] [1489098050.854526]: has_gyro False
/opt/ros/kinetic/lib/turtlebot_calibration/calibrate.py:79: SyntaxWarning: The publisher should be created with an explicit keyword argument 'queue_size'. Please see http://wiki.ros.org/rospy/Overview/Publishers%20and%20Subscribers for more information.
  self.cmd_pub = rospy.Publisher('cmd_vel', Twist)
Waiting for service /turtlebot_node/set_parameters...
[ INFO] [1489098053.321904353]: Starting depth stream.
[ INFO] [1489098053.441148566]: using default calibration URL
[ INFO] [1489098053.443462943]: camera calibration URL: file:///home/habals/.ros/camera_info/depth_Astra_Orbbec.yaml
[ INFO] [1489098053.446202508]: Unable to open camera calibration file [/home/habals/.ros/camera_info/depth_Astra_Orbbec.yaml]
[ WARN] [1489098053.448401437]: Camera calibration file /home/habals/.ros/camera_info/depth_Astra_Orbbec.yaml not found.

Originally posted by kangnamtofu with karma: 28 on 2017-03-09
This answer was ACCEPTED on the original site
Post score: 0

